### DISCUSSION TITLE
Brownie problem in contract

### DISCUSSION THREAD

**Discussion by oscarmairey** [smartcontractkit/full-blockchain-solidity-course-py#230] [Q&A] [answered]
Hi, I'm currently at the part5 of the tutorial using brownie.
I just installed it (successfully) as this screenshot states
![image](https://user-images.githubusercontent.com/92814105/138164970-257ab8f1-a3c4-4421-9153-a4790ab16b7a.png)

However, my script does not recognize brownie as a keyword 
![image](https://user-images.githubusercontent.com/92814105/138165193-42486712-fe23-41c6-a6dd-262e1e15728b.png)

the errorr 
reportMissingImports | Diagnostics for imports that have no corresponding imported python file or type stub file.
shows up
and brownie is perfectly installed 
what to do ?

Thank you so much for responses and taking your time helping out, I've been trying for half an hour now and I cant get an answer.

**Comment by cromewar** [comment, marked as answer]
I already answered this on this thread: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/209#discussioncomment-1499503

But the answer is Vscode python extension is incapable to recognize any package installed using `pipx`, the compiler will work anyway, but if you want to get rid of problem, install eth-brownie using `pip` instead, but have in mind you would like to use  a different virtual environment for each project.

Best regards.

**Reply by oscarmairey** [reply]
Hi, thank you for the time. I unsinstalled everything and installed using pip but it does not work. Any clues on why ? I have now both pip and pipx installed now

**Reply by cromewar** [reply]
Hi @Cesarioo.

Hint1: get rid off pipx, you could have dependency conflicts having both on your machine.
Hint2: remember to install `eth-brownie` instead of just `brownie`: `pip install eth-brownie`
Hint3: use a virtual environment for each project, here is a tutorial: https://help.dreamhost.com/hc/es/articles/115000695551-Instalar-y-usar-virtualenv-con-Python-3

Additionally could you please explain further what's the problem your are facing now?
